CrossPlatformDesktopProject/CrossPlatformDesktopProject/NPC/Npc.cs

```diff
@@ -0,0 +1,37 @@
+﻿using CrossPlatformDesktopProject;
+using CrossPlatformDesktopProject.NPC;
+using CrossPlatformDesktopProject.NPC.Gel;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CrossPlatformDesktopProject.NPC
+{
+    class Npc
+    {
```

Comment:
I think it's confusing that there's an `INpc` interface that `Npc` doesn't implement.

I wonder if it would be better to have an `INpc` interface implemented by a `Goriya` class, and `Goriya` objects store `IGoriyaState`s, where things like `GoriyaWalkSouth` implement `IGoriyaState`? Maybe this would give a nicer way to construct a `new Goriya()` by calling its constructor, rather than by constructing a general NPC.

Reply:
So do you mean like having a separate `Goriya` class from `GoriaWalkSouth` and etc.? And depending on `IGoriyaState` it will call the required direction? Sorry I'm a bit confused.

Reply:
I was thinking
```
interface INpc {...}

interface IGoriyaState {...}

class Goriya : INpc {
    public IGoriyaState currentState;
    ...
}

class GoriyaWalkSouth : IGoriyaState {
    
}

... [other Goriya states] ...
```

and then

```
interface IBatState {...}

class Bat : INpc
{
    public IBatState currentState;
    ...
}

class BatFlyingSouthOrWhatever : IBatState {...}

... [other Bat states] ...
```

I was just thinking that there's no reason a Goriya would have a Bat state, so making those interfaces only per-enemy type would be more enapsulated. I could be wrong though, so feel free to disagree.

Reply:
I just made the changes, hopefully I understood you right lol. If not, just let me know and I'll fix whatever. Sorry for taking so long, had an exam this morning that I was studying for.